scipy/stats/stats.py

```diff
@@ -366,6 +366,10 @@ def gmean(a, axis=0, dtype=None):
         dtype of a, unless a has an integer dtype with a precision less than
         that of the default platform integer. In that case, the default
         platform integer is used.
+    weights : array_like, optional
```

Comment:
can `weights` be negative or `0`, can they be any type or should they be an array_like of ints or positive floats?

Reply:
The calculation can definitively be performed with non-positive weights. Whether this actually makes sense, depends IMHO on the application of the user.

Reply:
ok thanks for confirming. Let's leave this as is then, marking resolved.